hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -839,7 +839,17 @@ int SaraNcpClient::waitReady(bool powerOn) {
         ready_ = waitAtResponseFromPowerOn(modemState) == 0;
     } else if (ncpState() == NcpState::OFF) {
         LOG_DEBUG(TRACE, "Waiting for modem to be ready from current unknown state");
-        ready_ = checkRuntimeState(modemState) == 0;
+        if (ncpId() != PLATFORM_NCP_SARA_R410) {
+            ready_ = checkRuntimeState(modemState, UBLOX_NCP_RUNTIME_SERIAL_BAUDRATE_U2) == 0;
+        } else {
+            int res = checkRuntimeState(modemState, UBLOX_NCP_RUNTIME_SERIAL_BAUDRATE_R4);
+            if (res == SYSTEM_ERROR_UNKNOWN) {
+                // resuming muxer has failed probably because of baud rate. Try with other baud rate
+                LOG_DEBUG(TRACE, "Checking muxer at %d baud", UBLOX_NCP_DEFAULT_SERIAL_BAUDRATE);
+                res = checkRuntimeState(modemState, UBLOX_NCP_DEFAULT_SERIAL_BAUDRATE);
```

Comment:
This will do extra steps which we don't really need:
1. Check muxer at runtime baudrate
2. Check at raw runtime baudrate
3. Check at raw default baudrate
4. Check muxer at default baudrate
5. Check at raw default baudrate
6. Check at raw default baudrate

Whereas the sequence should be:
1. Check muxer at runtime baudrate
2. Check muxer at default baudrate
3. Check at raw runtime baudrate
4. Check at default runtime baudrate

Reply:
This is still an extra call.